src/stp/computations/PositionComputations.cpp

```diff
@@ -31,111 +31,105 @@ Vector2 PositionComputations::getWallPosition(int index, int amountDefenders, co
     if (ComputationManager::calculatedWallPositions.empty()) {
         ComputationManager::calculatedWallPositions = determineWallPositions(field, world, amountDefenders);
     }
+
     return ComputationManager::calculatedWallPositions[index];
 }
 
 std::vector<Vector2> PositionComputations::determineWallPositions(const rtt::world::Field &field, const rtt::world::World *world, int amountDefenders) {
     if (amountDefenders <= 0) return {};  // we need at least 1 defender to be able to compute a wall
-    Vector2 ballPos = FieldComputations::projectPointInField(field, world->getWorld().value().getBall()->get()->getPos());
+
     double radius = control_constants::ROBOT_RADIUS;
-    double spacingRobots = radius * 2;
-    double spaceBetweenDefenseArea = field.getFieldLength() / 30;  // Because path planning is weird about being right next to a defense area
+    double spacingRobots = radius * 0.5;
+    double spaceBetweenDefenseArea = 2 * radius;
+
+    Vector2 ballPos = FieldComputations::projectPointInField(field, world->getWorld().value().getBall()->get()->getPos());
 
     std::vector<Vector2> positions = {};
-    Vector2 lineBorderIntersect;
+
+    Vector2 projectedPosition;
     std::vector<Vector2> lineBorderIntersects = {};
 
     /// Get defense area border geometry
-    std::vector<LineSegment> defenseAreaBorder = FieldComputations::getDefenseArea(field, true, radius + control_constants::GO_TO_POS_ERROR_MARGIN, 0).getBoundary();
+    std::vector<LineSegment> defenseAreaBorder = FieldComputations::getDefenseArea(field, true, spaceBetweenDefenseArea, 0).getBoundary();
 
     /// Find intersect of ball to goal on the border of the defense area
     LineSegment ball2GoalLine = LineSegment(ballPos, field.getOurGoalCenter());
-    for (const LineSegment &line : defenseAreaBorder) {  // Vector is made to check if there is only 1 intersect
-        if (line.doesIntersect(ball2GoalLine)) {
-            auto intersect = line.intersects(ball2GoalLine);
-            if (intersect.has_value() &&
-                // check if there is an intersect and that the intersect is not with the goal line
-                intersect->x - field.getOurGoalCenter().x > radius + control_constants::GO_TO_POS_ERROR_MARGIN)
-                lineBorderIntersects.push_back(intersect.value());
+    for (auto i : defenseAreaBorder) {
```

Comment:
Why would you go over all the edges of the defense area to check for intersections when the Polygon that is being returned from `getDefenseArea` already has an built-in function that does all of this for you already?